src/List.svelte

```diff
@@ -0,0 +1,48 @@
+<script lang="ts">
+  import Loading from "@app/Loading.svelte";
+
+  export let items: any[];
+  export let query: any;
+
+  // Being passed to more slot to handle loading state of trigger
+  let loading = false;
+  // Should be changed to true, once no more items are being returned.
+  let complete = false;
+
+  const fetchMore = async () => {
+    loading = true;
+    const response = await query();
+
+    if (response.length > 0) {
+      items.push(...response);
+      // We explicitly do an assignment here to force Svelte to update the slots props
+      items = items;
+    } else {
+      complete = true;
+    }
+
+    loading = false;
+  };
+</script>
+
+<style>
+  .more {
+    margin-top: 1rem;
+    text-align: center;
+  }
+</style>
+
+<slot name="list" {items} />
+{#if complete === false}
+  <slot name="more" {fetchMore} {loading}>
+    <div class="more">
+      <button class="small" on:click={fetchMore}>
+        {#if loading}
+          <Loading center small />
+        {:else}
+          More
+        {/if}
+      </button>
+    </div>
+    </slot>
```

Comment:
indent

Reply:
Done in [a23c649](https://github.com/radicle-dev/radicle-interface/pull/344/commits/a23c649bf5fd56a7a4bb830a47274adb72f97ba0)